File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        int ans[] = new int[k];
        HashMap<Integer, Integer> map = new HashMap<>();
        LinkedHashMap<Integer, Integer> sortedMap = new LinkedHashMap<>();
        for(int i=0; i<nums.length; i++){
            map.putIfAbsent(nums[i],0);
            map.put(nums[i], map.get(nums[i])+1);
        }
        ArrayList<Integer> list = new ArrayList<Integer>(map.values());
        Collections.sort(list, Collections.reverseOrder()); 
        for (int num : list) {
            for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
                if (entry.getValue().equals(num)) {
                    sortedMap.put(entry.getKey(), num);
                }
            }
        }
        Set<Integer> keySet = sortedMap.keySet();
        ArrayList<Integer> keyIndex = new ArrayList<>(keySet);
        for(int i=0; i<k; i++){
            ans[i] = keyIndex.get(i);
        }
        return ans;
    }
}

File: Data Structures & Algorithms/subsets-ii/submission-0.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        res = []
        nums.sort()
        def backtrack(ans,i):
            if i == len(nums):
                res.append(ans[:])
                return
            
            ans.append(nums[i])
            backtrack(ans,i+1)

            while i < len(nums) - 1 and nums[i] == nums[i+1]:
                i += 1
            
            ans.pop()
            backtrack(ans,i+1)
        backtrack([],0)
        return res
